File: js/src/check-lines.js
export function checkLines(board) {
  return board.reduce((lines, row, ri) => {
    if (row.every((col) => col)) lines.push(ri);
    return lines;
  }, []);
}

export function clearLines(board, lines) {
  lines.toReversed().forEach((ri) => {
    board.splice(ri, 1);
  });
  board.push(...Array.from({ length: lines.length }, () => Array(10).fill(0)));
}

File: python/tetris/check_lines.py
def check_lines(board):
    lines = []
    for i, row in enumerate(board):
        if 0 not in row:
            lines.append(i)
    return lines


def clear_lines(board, lines):
    for i in reversed(lines):
        del board[i]
        board.append([0] * 10)

    return board
